### DISCUSSION TITLE
How to make Tasmotized devices show up in the Home Assistant MQTT integration

### DISCUSSION THREAD

**Discussion by oywino** [arendst/Tasmota#19872] [Q&A] [answered]
I have number of Sonoff devices flashed with Tasmota 12.2.0 and configured to communicate with my Mosquitto MQTT broker.
They all work great and I can control them from Home Assistant by adding them as binary sensors in a dedicated MQTT section.
What puzzles me is; Why do these devices not show up as entities in the MQTT Integration?

I have several other (non-Tasmota) MQTT clients, and they all show up in the entity ist in the MQTT Integration in HA.

**Comment by barbudor** [comment]
As I'm not using HA, I'm not sure what you call 'mqtt integration' but there is definitively a plug-in that is required to benefit from Tasmota Discovery protocol
https://github.com/emontnemery/hatasmota

**Reply by sfromis** [reply]
You should not expect them to show up in MQTT integration, but instead in the replacement Tasmota integration, as pointed to by barbudor.

**Comment by oywino** [comment]
I installed this Tasmota Integration: https://www.home-assistant.io/integrations/tasmota
And it works OK and I can now see each device and its entities - great!
But one device is a Sonoff 433RF Bridge, and I cannot see the devices connected to it anywhere.

**Reply by barbudor** [reply]
Sonoff bridge does not work with Tasmota integration
Only native peripherals are reported by Tasmota Discovery (those connected to gpios of Tasmota)
Same with BLE and Zigbee: peripherals that are not directly attached to the ESP are not reported

**Comment by oywino** [comment]
I'm not sure what you mean.  My bridge does seem to work well with the Tasmota integration, and it shows up in HA like this:
![image](https://github.com/arendst/Tasmota/assets/18447119/f5671570-e896-4ae3-b035-a43dd9512145)

**Reply by barbudor** [reply]
You see the Tasmota device, but you complained not seeing the 433Mhz sensors or others. That what I explained

**Reply by sfromis** [reply, marked as answer]
Tasmota can only report devices it already knows, which excludes RF 433 stuff. And the reporting happens a boot, meaning only devices already recognized by Tasmota at boot time. Which again excludes RF 433 stuff. Thus HA can only see the bridge as such, not what you may be planning to push of buttons later.

In https://tasmota.github.io/docs/Home-Assistant/#binary-sensors the example "RF Bridge" shows how such can be manually configured in HA. Note that the example assumes that you also have set a RfKey; if not you can also use other fields from the MQTT message you get when pushing a button.
